src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -13109,5 +13161,512 @@ public static void Main()
+        }
+
+        [ConditionalFact(typeof(NoIOperationValidation))]
+        public void LambdaWithDefault_NonConstantLiteral_InterpolatedString()
+        {
+            var source = """
+class Program
+{
+    // Named delegate has required parameter x
+    public static int f(int x) => 2 * x;
+    public static void Main()
+    {
+        int n = 42;
+        // lambda has optional parameter x
+        var lam = (string s = $"n: {n}") => { };
+    }
+}
+""";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (9,31): error CS1736: Default parameter value for 's' must be a compile-time constant
+                //         var lam = (string s = $"n: {n}") => { };
+                Diagnostic(ErrorCode.ERR_DefaultValueMustBeConstant, @"$""n: {n}""").WithArguments("s").WithLocation(9, 31));
+        }
+
+        [Fact]
+        public void LambdaWithDefault_NonConstantLiteral_U8String()
+        {
+            TestDiagnosticsInMain(
+"""
+var lam = (ReadOnlySpan<byte> s = "u8 string"u8) => { };
+""", usings: new[] { "System" }, targetFramework: TargetFramework.Net60,
+                // (6,43): error CS1736: Default parameter value for 's' must be a compile-time constant
+                //         var lam = (ReadOnlySpan<byte> s = "u8 string"u8) => { };
+                Diagnostic(ErrorCode.ERR_DefaultValueMustBeConstant, @"""u8 string""u8").WithArguments("s").WithLocation(6, 43));
+
+        }
+
+        [Fact]
+        public void LambdaWithParameterDefaultValueAttribute()
+        {
+            var source = """
+using System;
+using System.Runtime.InteropServices;
+
+class Program
+{
+    static void Report(object obj) => Console.WriteLine(obj.GetType());
+    public static void Main()
+    {
+        var lam = ([DefaultParameterValue(3)] int x) => x;
+        Console.WriteLine(lam());
+        Report(lam);
+    }
+}
+""";
+            CompileAndVerify(source, expectedOutput:
+@"3
+<>f__AnonymousDelegate0").VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void LambdaDefaultParameter_UnsafeNull()
+        {
+            var source = """
+using System;
+
+class Program
+{
+    public static unsafe void Main()
+    {
+        var lam = (int *ptr = null) => ptr;
+        Console.WriteLine(lam() == (int*) null);
+    }
+}
+""";
+            CompileAndVerify(source, options: TestOptions.UnsafeReleaseExe, verify: Verification.Skipped, expectedOutput: "True").VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void LambdaDefaultParameter_UnsafeSizeof()
+        {
+            TestInMain("""
+unsafe
+{
+    var lam = (int sz = sizeof(int)) => sz;
+    Console.WriteLine(lam());
+}
+""", usings: new[] { "System" }, expectedOutput: "4", compilationOptions: TestOptions.UnsafeReleaseExe);
+        }
+
+        // PROTOTYPE: The emitted lambda method uses object as the parameter type, even though
```

Comment:
`dynamic` _is_ object, but with a special attribute applied to tell the compiler to treat it as `dynamic`. You can see it on line 13433.

Reply:
Yes, thanks. That's what I thought but wasn't sure (this was originally Adam's code and comment). Removed the comment: f9274a5d23b1c9e3d3f8b49537854039e804df1b

I have also noticed another PROTOTYPE comment that I believe is completed now, so I removed it, as well: cffc52ec183fa09def54bb0db6e770f0635429f9